client/modules/IDE/components/Editor/index.jsx

```diff
@@ -146,168 +336,135 @@ class Editor extends React.Component {
       }
     });
 
-    this.hinter = new Fuse(hinter.p5Hinter, {
-      threshold: 0.05,
-      keys: ['text']
-    });
-
-    delete this._cm.options.lint.options.errors;
+    const cm = cmInstance.current;
+    emmet(CodeMirror);
 
     const replaceCommand =
       metaKey === 'Ctrl' ? `${metaKey}-H` : `${metaKey}-Option-F`;
-    this._cm.setOption('extraKeys', {
-      Tab: (cm) => {
-        if (!cm.execCommand('emmetExpandAbbreviation')) return;
-        // might need to specify and indent more?
-        const selection = cm.doc.getSelection();
+    cm.setOption('extraKeys', {
+      Tab: (_cmInstance) => {
+        if (!_cmInstance.execCommand('emmetExpandAbbreviation')) return;
+        const selection = _cmInstance.doc.getSelection();
         if (selection.length > 0) {
-          cm.execCommand('indentMore');
+          _cmInstance.execCommand('indentMore');
         } else {
-          cm.replaceSelection(' '.repeat(INDENTATION_AMOUNT));
+          _cmInstance.replaceSelection(' '.repeat(INDENTATION_AMOUNT));
         }
       },
       Enter: 'emmetInsertLineBreak',
       Esc: 'emmetResetAbbreviation',
       [`${metaKey}-Enter`]: () => null,
       [`Shift-${metaKey}-Enter`]: () => null,
       [`${metaKey}-F`]: 'findPersistent',
-      [`Shift-${metaKey}-F`]: this.tidyCode,
+      [`Shift-${metaKey}-F`]: tidyCode,
       [`${metaKey}-G`]: 'findPersistentNext',
       [`Shift-${metaKey}-G`]: 'findPersistentPrev',
       [replaceCommand]: 'replace',
-      // Cassie Tarakajian: If you don't set a default color, then when you
-      // choose a color, it deletes characters inline. This is a
-      // hack to prevent that.
-      [`${metaKey}-K`]: (cm, event) =>
-        cm.state.colorpicker.popup_color_picker({ length: 0 }),
-      [`${metaKey}-.`]: 'toggleComment' // Note: most adblockers use the shortcut ctrl+.
+      [`${metaKey}-K`]: (_cmInstMetaKey, event) =>
```

Comment:
can we keep the comments from cassie and other notes? will probably be nice for the future : )